### PR TITLE
Additional parameters 'Label' and 'Closed' added to TAggrResampler

### PR DESCRIPTION
**Pull request by bkazic**
- `Closed`: Which side of bin interval is closed. The default is ‘left’.
- `Label`: Which bin edge label to label bucket with. The default is ‘left’.

Ref: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.resample.html#pandas.DataFrame.resample

### CODE REVIEW

File: src/glib/mine/signalproc.cpp [hunk 1]
```diff
@@ -1007,7 +1017,8 @@ bool TAggrResampler::TryResampleOnce(double& Val, uint64& Tm, bool& FoundEmptyP)
         // interval [LastResampPointMSecs + IntervalMSecs , LastResampPointMSecs + 2 * IntervalMSecs)
```
Comment (rupnikj) [inline src/glib/mine/signalproc.cpp]: comment fix

File: src/glib/mine/signalproc.cpp [hunk 2]
```diff
@@ -907,14 +907,22 @@ TAggrResampler::TAggrResampler(const PJsonVal& ParamVal) {
     }
     // XXX decide if we should use NaN for default (max,min,avg are not defined for empty buffers)?
     DefaultVal = ParamVal->GetObjNum("defaultValue", 0);
+    Closed = ParamVal->GetObjStr("closed", "left");
+    EAssertR(Closed == "left" || Closed == "right", "TAggrResampler: closed should be 'left' or 'right'");
+    Label = ParamVal->GetObjStr("label", "left");
+    EAssertR(Closed == "left" || Closed == "right", "TAggrResampler: label should be 'left' or 'right'");
```
Comment (rupnikj) [inline src/glib/mine/signalproc.cpp]: copy paste

File: src/glib/mine/signalproc.cpp [hunk 3]
```diff
@@ -1007,7 +1017,8 @@ bool TAggrResampler::TryResampleOnce(double& Val, uint64& Tm, bool& FoundEmptyP)
         // interval [LastResampPointMSecs + IntervalMSecs , LastResampPointMSecs + 2 * IntervalMSecs)
         // assert that we have not found a point older than the current window
         uint64 Start = LastResampPointMSecs + IntervalMSecs;
-        uint64 End = LastResampPointMSecs + 2 * IntervalMSecs - 1;
+        uint64 End = LastResampPointMSecs + 2 * IntervalMSecs;
+        int Edge = (Closed == "right") ? 0 : 1;
         while (!Buff.Empty()) {
             const TUInt64FltPr& Point = Buff.Top();
             if (Point.Val1 < Start) {
```
Comment (rupnikj) [inline src/glib/mine/signalproc.cpp]: + Edge

File: src/glib/mine/signalproc.h [hunk 4]
```diff
@@ -773,6 +773,10 @@ class TAggrResampler {
     TStr RoundStart;
     /// Default value when the buffer is empty
     TFlt DefaultVal;
+    /// Which side of bin interval is closed. The default is ‘left’.
+    TStr Closed;
+    /// Which bin edge label to label bucket with. The default is ‘left’.
```
Comment (rupnikj) [inline src/glib/mine/signalproc.h]: Either `left` or `right`. Resampled time-series time points can correspond to bucket start points (`left`) or bucket end points (`right`).

File: src/glib/mine/signalproc.h [hunk 5]
```diff
@@ -773,6 +773,10 @@ class TAggrResampler {
     TStr RoundStart;
     /// Default value when the buffer is empty
     TFlt DefaultVal;
+    /// Which side of bin interval is closed. The default is ‘left’.
```
Comment (rupnikj) [inline src/glib/mine/signalproc.h]: Bucket intervals are half open. If Closed=`left`, the interval includes the left time point and is open on the right side (and the oposite if Closed=`right`).

**Review by bergloman — APPROVED** [review]

**Pull request merged**